### PR TITLE
feat: automates package installation

### PR DESCRIPTION
**Pull request by Jai0401**
## Description

Implement a mechanism to automatically prompt users to install missing packages. If they agree, the required package will be installed.

- progresses #1768 

Updated files:
`mem0/configs/vector_stores/chroma.py`
`mem0/llms/aws_bedrock.py`
`mem0/llms/groq.py`
`mem0/llms/litellm.py`
`mem0/llms/ollama.py`
`mem0/llms/together.py`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [ ] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Review by Dev-Khant — CHANGES_REQUESTED** [review]
Hey @Jai0401 Please avoid creating the functions here for checking if the package has been installed or not. You can directly use the code snippet that you provided in issue description.

**Comment by Jai0401** [comment]
@Dev-Khant I've made the changes requested in the previous review. Could you please take a look and let me know if everything is in order or if any further adjustments are needed?

File: mem0/configs/vector_stores/chroma.py [hunk 1]
```diff
@@ -7,9 +9,19 @@ class ChromaDbConfig(BaseModel):
     try:
```
Comment (Dev-Khant) [inline mem0/configs/vector_stores/chroma.py:9]: Can you please move this import statement out from the class.

**Review by Dev-Khant — COMMENTED** [review]
Also can you please make changes from groq, litellm, together and ollama llms.

**Comment by Jai0401** [comment]
@Dev-Khant I've made the requested changes. Please review them and let me know if any further adjustments are needed.

File: mem0/configs/vector_stores/chroma.py [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+        except subprocess.CalledProcessError:
+            print("Failed to install 'chromadb'. Please install it manually using 'pip install chromadb'.")
+            sys.exit(1)
+    else:
+        print("The required 'chromadb' library is not installed.")
+        sys.exit(1)
+
 
 class ChromaDbConfig(BaseModel):
-    try:
```
Comment (Dev-Khant) [inline mem0/configs/vector_stores/chroma.py:7]: Hey @Jai0401 Really sorry my bad, can you please keep the import statement inside the class.

**Review by Dev-Khant — COMMENTED** [review]
Rest looks good, left a small comment for `chromadb`.

**Comment by Jai0401** [comment]
@Dev-Khant  I've made the requested changes.

**Review by Dev-Khant — APPROVED** [review]

**Pull request merged**